jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java

```diff
@@ -324,62 +336,67 @@ public void destroy() {
-        } else if (pingTask.isRunning() && pingTask.isException()) {
-            // Reset ping task if it is throwing exceptions
-            pingTask = new PingTask(session);
-            resetPingTask = true;
-        }
-
-        if (resetPingTask) {
-            if (pingScheduledExecutorService != null) {
-                pingScheduledExecutorService.shutdownNow();
-            }
-
-            pingScheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
-            pingScheduledExecutorService.scheduleAtFixedRate(pingTask, 1L, 30L, TimeUnit.SECONDS);
-        }
-    }
-
-    class PingTask implements Runnable {
-
-        WebSocketSession webSocketSession;
-        boolean isRunning;
-        boolean isException;
-
-        PingTask(WebSocketSession webSocketSession) {
-            this.webSocketSession = webSocketSession;
-        }
-
-        @Override
-        public void run() {
-            try {
-                logger.debug("Pinging Slack...");
-                isRunning = true;
-                Message message = new Message();
-                message.setType(EventType.PING.name().toLowerCase());
-                synchronized (sendMessageLock) {
-                    webSocketSession.sendMessage(new TextMessage(message.toJSONString()));
-                }
-                isException = false;
-            } catch (Exception e) {
-                logger.error("Error pinging Slack. Slack bot may go offline when not active. Exception: ", e);
-                isException = true;
-            }
-        }
-
-        boolean isRunning() {
-            return isRunning;
-        }
-
-        boolean isException() {
-            return isException;
-        }
-    }
+	private void pingAtRegularIntervals(WebSocketSession session) {
+		// only get called on hello response (connect)
+		pingTask = new PingTask(session);
+		if (pingScheduledExecutorService != null) {
+			pingScheduledExecutorService.shutdownNow();
+		}
+		pingScheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+		pingScheduledExecutorService.scheduleAtFixedRate(pingTask, 1L, 30L, TimeUnit.SECONDS);
+
+	}
+
+	class PingTask implements Runnable {
+
+		WebSocketSession webSocketSession;
+		boolean isRunning;
+		boolean isException;
+
+		PingTask(WebSocketSession webSocketSession) {
+			this.webSocketSession = webSocketSession;
+		}
+
+		@Override
+		public void run() {
+			try {
+				logger.debug("Pinging Slack...");
+				isRunning = true;
+				Message message = new Message();
+				message.setType(EventType.PING.name().toLowerCase());
+				synchronized (sendMessageLock) {
+					webSocketSession.sendMessage(new TextMessage(message.toJSONString()));
+				}
+				isException = false;
+			} catch (Exception e) {
+				logger.error("Error pinging Slack. Slack bot may go offline when not active. Exception: ", e);
+				isException = true;
+				if (!isOpen()) {
+					try {
+						manager.stop();
+					} catch (Throwable t) {
```

Comment:
Change to
```suggestion
					} catch (Exception e) {
```
See [this](https://stackoverflow.com/a/581889/1385441) for more info.